pkg/packagekit/render_launchd.go

```diff
@@ -0,0 +1,62 @@
+package packagekit
+
+import (
+	"fmt"
+	"io"
+
+	"github.com/groob/plist"
+	"github.com/pkg/errors"
+)
+
+// Note: I wanted to just include the InitOptions struct here, but it
+// did not behave as expected. plist.Encode printed the fields without
+// hinting
+type launchdOptions struct {
+	Environment       map[string]string      `plist:"EnvironmentVariables"`
+	Args              []string               `plist:"ProgramArguments"`
+	Label             string                 `plist:"Label"`
+	ThrottleInterval  int                    `plist:"ThrottleInterval"`
+	StandardErrorPath string                 `plist:"StandardErrorPath"`
+	StandardOutPath   string                 `plist:"StandardOutPath"`
+	KeepAlive         map[string]interface{} `plist:"KeepAlive"`
+}
+
+type lOption func(*launchdOptions)
```

Comment:
this is private, but RenderLaunchd is public. decide on one. IMO `renderLaunchd` should be private, there's arguably no need to call it independently.

Reply:
I think this is different now. I took a stab using https://derekchiang.com/posts/reusable-and-type-safe-options-for-go-apis/